core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala

```diff
@@ -45,103 +52,107 @@ class DeleteTopicTest extends QuorumTestHarness {
-    adminZkClient.deleteTopic(topic)
-    TestUtils.verifyTopicDeletion(zkClient, topic, 1, servers)
+    createTestTopicAndCluster(topic)
+    TestUtils.deleteTopicWithAdmin(adminClient, topic, brokers)
   }
 
-  @Test
-  def testResumeDeleteTopicWithRecoveredFollower(): Unit = {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testResumeDeleteTopicWithRecoveredFollower(quorum: String): Unit = {
     val topicPartition = new TopicPartition("test", 0)
     val topic = topicPartition.topic
-    servers = createTestTopicAndCluster(topic)
+    createTestTopicAndCluster(topic)
     // shut down one follower replica
-    val leaderIdOpt = zkClient.getLeaderForPartition(new TopicPartition(topic, 0))
-    assertTrue(leaderIdOpt.isDefined, "Leader should exist for partition [test,0]")
-    val follower = servers.filter(s => s.config.brokerId != leaderIdOpt.get).last
+    val partition = getTopicPartitionInfo(adminClient, topic, 0)
+    assertTrue(partition.isPresent, "Partition [test,0] should exist")
+    val leaderNode = partition.get().leader()
+    assertFalse(leaderNode.isEmpty, "Leader should exist for partition [test,0]")
+    val follower = brokers.filter(s => s.config.brokerId != leaderNode.id()).last
     follower.shutdown()
     // start topic deletion
-    adminZkClient.deleteTopic(topic)
+    faultHandler.setIgnore(true)
+    adminClient.deleteTopics(Collections.singleton(topic))
     // check if all replicas but the one that is shut down has deleted the log
     TestUtils.waitUntilTrue(() =>
-      servers.filter(s => s.config.brokerId != follower.config.brokerId)
-        .forall(_.getLogManager.getLog(topicPartition).isEmpty), "Replicas 0,1 have not deleted log.")
+      brokers.filter(s => s.config.brokerId != follower.config.brokerId)
+        .forall(_.logManager.getLog(topicPartition).isEmpty), "Replicas 0,1 have not deleted log.")
     // ensure topic deletion is halted
-    TestUtils.waitUntilTrue(() => zkClient.isTopicMarkedForDeletion(topic),
-      "Admin path /admin/delete_topics/test path deleted even when a follower replica is down")
+    val aliveBrokers = brokers.filter(b => b != follower)
+    TestUtils.waitForAllPartitionsMetadata(aliveBrokers, topic, 0)
+    TestUtils.waitForAllPartitionsMetadata(Seq(follower), topic, 1)
+    assertThrows(classOf[ExecutionException], () => adminClient.describeTopics(Collections.singleton(topic)).topicNameValues().get(topic).get())
     // restart follower replica
     follower.startup()
-    TestUtils.verifyTopicDeletion(zkClient, topic, 1, servers)
+    TestUtils.waitForAllPartitionsMetadata(brokers, topic, 0)
   }
 
-  @Test
-  def testResumeDeleteTopicOnControllerFailover(): Unit = {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testResumeDeleteTopicOnControllerFailover(quorum: String): Unit = {
```

Comment:
I feel this test and some of the others are not super relevant to kraft. In the zookeeper-based implementation, we have logic to delay deletion until all replicas acknowledge, but there is nothing similar in kraft. So all of the tests which try to get the topic into a partially deleted state do not really work the same in kraft. For kraft, the more interesting scenarios to test involve recreation of the topic while the replica is offline. For example, do we have a test covering a case like this:

1. Create partition with two replicas: 0, 1
2. Shutdown broker 0
3. Delete the topic 
4. Recreate the topic
5. Restart broker 0 
6. Verify that the right topicId is stored on disk on brokers 0 and 1